tests/unit/lib/positions/init_spec.lua

```diff
@@ -0,0 +1,344 @@
+      },
+    }
+    assert.are.same(expected, ret:to_list())
+  end)
+
+  it("merge(dir, file) adds new file", function()
+    local dir = create_tree({
+      { type = "dir", id = "/root", path = "/root" },
+    })
+    local file = create_tree({
+      { type = "file", id = "/root/file", path = "/root/file" },
+      { type = "test", id = "test-2", path = "/root/file" },
+    })
+    local ret = positions.merge(dir, file)
+    local expected = {
+      {
+        id = "/root",
+        path = "/root",
+        type = "dir",
+      },
+      {
+        {
+          id = "/root/file",
+          path = "/root/file",
+          type = "file",
+        },
+        {
+          {
+            id = "test-2",
+            path = "/root/file",
+            type = "test",
+          },
+        },
+      },
+    }
+    assert.are.same(expected, ret:to_list())
+  end)
+
+  it("merge(file, file) updates tests in file", function()
+    local file = create_tree({
+      { type = "file", id = "file", path = "/root/file" },
+      { { type = "test", id = "test-1", path = "/root/file" } },
+    })
+    local new_file = create_tree({
+      { type = "file", id = "file", path = "/root/file" },
+      { { type = "test", id = "test-2", path = "/root/file" } },
+    })
+    local ret = positions.merge(file, new_file)
+    local expected = {
+      {
+        id = "file",
+        path = "/root/file",
+        type = "file",
+      },
+      {
+        {
+          id = "test-2",
+          path = "/root/file",
+          type = "test",
+        },
+      },
+    }
+    assert.are.same(expected, ret:to_list())
+  end)
+
+  it("merge(dir, dir) merges unrelated dirs at common ancestor", function()
+    local dir = create_tree({
+      { type = "dir", id = "/root/sub1", path = "/root/sub1" },
+      {
+        { type = "file", id = "/root/sub1/file", path = "/root/sub1/file" },
+        { { type = "test", id = "test-1", path = "/root/sub1/file" } },
+      },
+    })
+    local new_dir = create_tree({
+      { type = "dir", id = "/root/sub2", path = "/root/sub2" },
+      {
+        { type = "file", id = "/root/sub2/file", path = "/root/sub2/file" },
+        { { type = "test", id = "test-2", path = "/root/sub2/file" } },
+      },
+    })
+    local ret = positions.merge(dir, new_dir)
+    local expected = {
+      { type = "dir", id = "/root", name = "root", path = "/root" },
+      {
+        { type = "dir", id = "/root/sub1", path = "/root/sub1" },
+        {
+          { type = "file", id = "/root/sub1/file", path = "/root/sub1/file" },
+          { { type = "test", id = "test-1", path = "/root/sub1/file" } },
+        },
+      },
+      {
+        { type = "dir", id = "/root/sub2", path = "/root/sub2" },
+        {
+          { type = "file", id = "/root/sub2/file", path = "/root/sub2/file" },
+          { { type = "test", id = "test-2", path = "/root/sub2/file" } },
+        },
+      },
+    }
+    print(string.format("ret: %s", vim.inspect(ret:to_list())))
```

Comment:
Rogue print :sweat_smile:

Reply:
Dangit! I need to add a precommit hook for these 😅